core/client/hdfs/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -211,6 +212,22 @@ public void hadoopShouldLoadFileSystemWhenConfigured() throws Exception {
     }
   }
 
+  /**
+   * Ensures that Hadoop loads the Alluxio file system with Zookeeper URI.
+   */
+  @Test
+  public void hadoopShouldLoadFileSystemWithZkUri() throws Exception {
+    org.apache.hadoop.conf.Configuration conf = getConf();
+
+    URI uri = URI.create(Constants.HEADER
+        + HadoopUtils.ZOOKEEPER_IDENTIFIER + "zkHost:2181/tmp/path.txt");
+
+    try (Closeable c = new ConfigurationRule(new HashMap<>()).toResource()) {
+      final org.apache.hadoop.fs.FileSystem fs = org.apache.hadoop.fs.FileSystem.get(uri, conf);
+      assertTrue(fs instanceof FileSystem);
```

Comment:
can we test that the zk is actually being picked up?

Reply:
Now I add the zookeeper information from URI into hadoop configuration and then merge hadoop configuration, so we could test if the zk is actually being picked up. Actually, it make more sense to add the Zookeeper information into Alluxio configuration directly, but then we are not able to test zk information here.